mission-7/queries.sql

```diff
@@ -0,0 +1,126 @@
+/*
+  다음 경우들에 대해 총 14개의 SQL 쿼리를 작성해 주세요.
+  예시로 값이 필요한 경우 적당한 값으로 채워넣어서 작성하면 됩니다. 
+*/
+
+/*
+  1. 내 정보 업데이트 하기
+  - 닉네임을 "test"로 업데이트
+  - 현재 로그인한 유저 id가 1이라고 가정
+*/
+-- INSERT INTO "User" (email, nickname, password) VALUES ('test@test.com', 'user1', '12345678a');
+UPDATE "User" SET nickname = 'test' WHERE id = 1;
+-- SELECT * from "User";
+
+/*
+  2. 내가 생성한 상품 조회
+  - 현재 로그인한 유저 id가 1이라고 가정
+  - 최신 순으로 정렬
+  - 10개씩 페이지네이션, 3번째 페이지
+*/
+-- INSERT INTO "Product" (name, description, price, tags, "userId") VALUES ('test 상품', '테스트입니다.', 10000, ARRAY['test1', 'test2'], 2);
+SELECT * FROM "Product" WHERE "userId" = 1 LIMIT 10 OFFSET 20;
+-- SELECT * from "Product";
+
+/*
+  3. 내가 생성한 상품의 총 개수
+  - 현재 로그인한 유저 id가 1이라고 가정
+*/
+SELECT COUNT(*) AS "총 개수" FROM "Product" WHERE "userId" = 1; 
+
+/*
+  4. 내가 좋아요 누른 상품 조회
+  - 현재 로그인한 유저 id가 1이라고 가정
+  - 최신 순으로 정렬
+  - 10개씩 페이지네이션, 3번째 페이지
+*/
+SELECT p.* 
+FROM "Product" as p
+JOIN "ProductLike" as pl
+on p.id = pl."productId"
+WHERE pl."userId" = 1
+ORDER BY pl."createdAt" DESC
+LIMIT 10
+OFFSET 20;
+
+/*
+  5. 내가 좋아요 누른 상품의 총 개수
+  - 현재 로그인한 유저 id가 1이라고 가정
+*/
+SELECT COUNT(p.*) 
+FROM "Product" as p
+JOIN "ProductLike" as pl
+on p.id = pl."productId"
+WHERE pl."userId" = 1;
+
+/*
+  6. 상품 생성
+  - 현재 로그인한 유저 id가 1이라고 가정
+*/
+INSERT INTO "Product" (name, description, price, tags, "userId") VALUES ('테스트 상품', '테스트입니다.', 50000, ARRAY['test1', 'test2'], 1);
+
+
+/*
+  7. 상품 목록 조회
+  - 상품명에 "test"가 포함된 상품 검색
+  - 최신 순으로 정렬
+  - 10개씩 페이지네이션, 1번째 페이지
+  - 각 상품의 좋아요 개수를 포함해서 조회하기
+*/
+SELECT p.*, count(pl."productId") as "좋아요 수"
+FROM "Product" as p
+LEFT JOIN "ProductLike" as pl
+ON p.id = pl."productId"
+WHERE p.name Like '%test%'
+GROUP BY p.id
+ORDER BY p."createdAt" DESC
+LIMIT 10;
+
+/*
+  8. 상품 상세 조회
+  - 1번 상품 조회
+*/
+SELECT * FROM "Product" WHERE id = 1;
+
+/*
+  9. 상품 정보 수정
+  - 1번 상품 수정
+*/
+UPDATE "Product" SET name = 'test 상품 수정' WHERE id = 1;
+
+/*
+  10. 상품 삭제
+  - 1번 상품 삭제
+*/
+DELETE FROM "Product" WHERE id = 1;
+
+/*
+  11. 상품 좋아요
+  - 1번 유저가 2번 상품 좋아요
+*/
+INSERT INTO "ProductLike" ("userId", "productId") VALUES (1, 2);
+
+/*
+  12. 상품 좋아요 취소
+  - 1번 유저가 2번 상품 좋아요 취소
+*/
+DELETE FROM "ProductLike" WHERE "userId" = 1 and "productId" = 2;
+
+/*
+  13. 상품 댓글 작성
+  - 1번 유저가 2번 상품에 댓글 작성
+*/
+INSERT INTO "Comment" (content, "userId", "productId") VALUES ('테스트 댓글 입니다.', 1, 2);
+
+/*
+  14. 상품 댓글 조회
+  - 1번 상품에 달린 댓글 목록 조회
+  - 최신 순으로 정렬
+  - 댓글 날짜 2025-03-25 기준일을 제외한 이전 데이터 10개
+*/
+SELECT *
+FROM "Comment"
+WHERE "productId" = 1 and "createdAt" < '2025-03-25'
+ORDER BY "createdAt" DESC
+LIMIT 10;
```

Comment:
단일 코멘트 테이블일 경우, 명시적으로 articleId = NULL을 조건에 줘서 선택도를 높일 수 있어요.

```suggestion
SELECT *
FROM "Comment"
WHERE "productId" = 1
  AND "articleId" IS NULL
  AND "createdAt" < TIMESTAMP '2025-03-25'
ORDER BY "createdAt" DESC, id DESC
LIMIT 10;
```